### PR TITLE
Added example for Matrix.rref() optional parameter iszerofunc

### LINKED ISSUE #12006: Matrix.rref() needs an improved docstring
This is the docstring:

```
Signature: M.rref(iszerofunc=<function _iszero at 0x7fa677a24c80>, simplify=False)
Docstring:
Return reduced row-echelon form of matrix and indices of pivot vars.

To simplify elements before finding nonzero pivots set simplify=True
(to use the default SymPy simplify function) or pass a custom
simplify function.

Examples
========

>>> from sympy import Matrix
>>> from sympy.abc import x
>>> m = Matrix([[1, 2], [x, 1 - 1/x]])
>>> m.rref()
(Matrix([
[1, 0],
[0, 1]]), [0, 1])
>>> rref_matrix, rref_pivots = m.rref()
>>> rref_matrix
Matrix([
[1, 0],
[0, 1]])
>>> rref_pivots
[0, 1]
```

Some points:

1. `iszerofunc` is not explained at all.
2. `simplify` is set to False, yet simplify will possibly be run. This is confusing.
3. There are no example for the different forms of the kwargs.

### PR DESCRIPTION
**Pull request by donaldwilson8**
#### References to other Issues or PRs

Partially addresses #12006

#### Brief description of what is fixed or changed
Added example of how the optional parameter iszerofunc can be used to avoid floating point errors in matrix reduction.

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/matrices/reductions.py [hunk 1]
```diff
@@ -288,6 +288,22 @@ def _rref(M, iszerofunc=_iszero, simplify=False, pivots=True,
     >>> rref_pivots
     (0, 1)
 
+    iszerofunc can correct rounding errors in matrices with float values.
+    In the following example, calling rref leads to floating point errors,
+    incorrectly row reducing the matrix. iszerofunc can set sufficiently
+    small numbers to zero and avoid this error.
```
Comment (asmeurer) [inline sympy/matrices/reductions.py:294]: There should be a blank line between the text and the example.

Comment (wilds-23) [reply sympy/matrices/reductions.py:294]: Alright, I just added the line.

File: .mailmap [hunk 2]
```diff
@@ -1329,4 +1330,4 @@ zzj <29055749+zjzh@users.noreply.github.com>
 Óscar Nájera <najera.oscar@gmail.com>
 Øyvind Jensen <jensen.oyvind@gmail.com>
 Łukasz Pankowski <lukpank@o2.pl>
-彭于斌 <1931127624@qq.com>
+彭于斌 <1931127624@qq.com>
```
Comment (oscargus) [inline .mailmap:1333]: Please do not remove the trailing newline.

File: sympy/matrices/reductions.py (hunk 1 above)
Comment (oscargus) [reply sympy/matrices/reductions.py:294]: Consider using formatting to make the text render more clearly. See https://docs.sympy.org/dev/guides/contributing/documentation-style-guide.html

Comment (wilds-23) [reply sympy/matrices/reductions.py]: I believe I made the correct formatting changes.

**Pull request merged**